test/manipulation.test.js

```diff
@@ -397,6 +677,27 @@ describe('manipulation', function () {
             });
         });
     });
+
+    it('should preserve properties with "undefined" value (promise variant)', function(done) {
+      Person.create({ name: 'b-name', gender: undefined })
+      .then(function(instance) {
+        instance.toObject().should.have.properties({
+          id: instance.id,
+          name: 'b-name',
+          gender: undefined
+        });
+
+        return Person.updateOrCreate({ id: instance.id, name: 'updated name 2' })
+        .then(function(updated) {
+          updated.toObject().should.have.properties({
+            id: instance.id,
+            name: 'updated name 2',
+            gender: undefined
+          });
+          done();
+        });
+      }).catch(done);
+    });
```

Comment:
"preserve properties with undefined value" is not related to callback vs. promise, can we omit this test?